server/src/main/java/com/linecorp/centraldogma/server/internal/storage/repository/Repository.java

```diff
@@ -94,10 +97,57 @@ default Author author() {
     }
 
     /**
-     * Validates the specified {@link Revision} and converts it into an absolute {@link Revision}.
+     * Returns the {@link CompletableFuture} whose value is the absolute {@link Revision} of the
+     * specified {@link Revision}. This will validate the specified {@link Revision} before converting it.
+     *
+     * @see #normalizeNow(Revision)
      */
     CompletableFuture<Revision> normalize(Revision revision);
 
+    /**
+     * Returns the absolute {@link Revision} of the specified {@link Revision}
+     * This will validate the specified {@link Revision} before converting it.
+     *
+     * @throws RevisionNotFoundException it the specified {@link Revision} is not found
+     *
+     * @see #normalize(Revision)
+     */
+    Revision normalizeNow(Revision revision);
+
+    /**
+     * Returns a {@link NormalizedFromToRevision} which contains an absolute {@code from} {@link Revision} and
+     * an absolute {@code to} {@link Revision}.
+     *
+     * @throws RevisionNotFoundException it the specified {@code from} or {@code to} is not found
+     */
+    default NormalizedFromToRevision normalizeFromToRevision(Revision from, Revision to) {
+        final Revision normalizedFrom = normalizeNow(requireNonNull(from, "from"));
+        final Revision normalizedTo = normalizeNow(requireNonNull(to, "to"));
+
+        return new NormalizedFromToRevision(normalizedFrom, normalizedTo, false);
+    }
+
+    /**
+     * Returns a {@link NormalizedFromToRevision} which contains an absolute {@code from} {@link Revision} and
+     * an absolute {@code to} {@link Revision}. If the specified {@code ascending} is {@code true},
+     * the {@link Revision#major} value of the {@code from} in the {@link NormalizedFromToRevision} will be
+     * lower or equal to the {@link Revision#major} value of the {@code to} in the
+     * {@link NormalizedFromToRevision}.
```

Comment:
How about:

- Removing `ascending` parameter and returning without switching `from` and `to`
- Adding the following methods to `NormalizedFromToRevision`:
  - `toAscending()`
  - `toDescending()`

..? (Because a user might want to preserve the original ordering)